pymc/sampling/forward.py

```diff
@@ -795,10 +795,17 @@ def sample_posterior_predictive(
     constant_coords = set()
     for dim, coord in trace_coords.items():
         current_coord = model.coords.get(dim, None)
+        current_length = model.dim_lengths.get(dim, None)
+        if hasattr(current_length, 'eval'):
+            current_length = current_length.eval()
```

Comment:
It can only be a shared variable in which case you can retrieve with `get_value()`, or a Constant in which case you can retrieve with `.data` That should be much more efficient than `eval` unless `eval` does something clever under the hood